### PR TITLE
Remove unused references to absl::Status (aka grpc_error_handle) and absl::StatusOr<>

### PR DESCRIPTION
**Pull request by gankineri**
Removing a number of unused variables. This has no behaviour change. 

These types are not considered "unused variables" by normal `-Wunused-variable` flags because they have nontrivial destructors, but these types' destructors are not used for their side effects, so unused variables of these types should be considered bug-prone. 

This PR removes all unused `absl::Status` and `absl::StatusOr<>` variables I could find in grpc.

### CODE REVIEW

**Comment by eugeneo** [comment]
@gankineri Thank you for your contribution! Please sign the CLA before our team can review the code.

File: src/core/ext/filters/client_channel/lb_policy/xds/cds.cc [hunk 1]
```diff
@@ -529,7 +529,7 @@ void CdsLb::OnClusterChanged(const std::string& name,
       gpr_log(GPR_INFO, "[cdslb %p] generated config for child policy: %s",
               this, json_str.c_str());
     }
-    grpc_error_handle error;
+
```
Comment (eugeneo) [inline src/core/ext/filters/client_channel/lb_policy/xds/cds.cc:532]: nit: Please remove the empty line.

File: src/core/ext/xds/xds_client_grpc.cc [hunk 2]
```diff
@@ -225,7 +225,7 @@ void SetXdsFallbackBootstrapConfig(const char* config) {
 grpc_slice grpc_dump_xds_configs(void) {
   grpc_core::ApplicationCallbackExecCtx callback_exec_ctx;
   grpc_core::ExecCtx exec_ctx;
-  grpc_error_handle error;
+
```
Comment (eugeneo) [inline src/core/ext/xds/xds_client_grpc.cc:228]: nit: please do not add an empty line.

File: src/core/lib/event_engine/posix_engine/posix_engine_listener_utils.cc [hunk 3]
```diff
@@ -137,7 +137,7 @@ int GetMaxAcceptQueueSize() {
 absl::Status PrepareSocket(const PosixTcpOptions& options,
                            ListenerSocket& socket) {
   ResolvedAddress sockname_temp;
-  absl::Status error;
+
```
Comment (eugeneo) [inline src/core/lib/event_engine/posix_engine/posix_engine_listener_utils.cc:140]: nit: please do not add an empty line.

**Review by eugeneo — CHANGES_REQUESTED** [review]
Thank you for your contribution. There are a bunch of minor code style nits.

**Comment by gankineri** [comment]
Thank you for the review, dropped the extra newlines.

**Review by markdroth — APPROVED** [review]
This looks good.  I've triggered the tests, so we'll see how things look.

**Comment by markdroth** [comment]
Looks like there are a bunch of test failures here.  I've sent you https://github.com/gankineri/grpc/pull/1 to fix them.  If you merge that PR, it will update this branch.

**Pull request merged**